sx126x/spi_stm32wl.go

```diff
@@ -0,0 +1,106 @@
+//go:build stm32wlx
+// +build stm32wlx
+
+package sx126x
+
+import (
+	"device/stm32"
+	"errors"
+	"machine"
+)
+
+// New creates a new SX126x connection.
+func New(spi machine.SPI) *Device {
+	c := make(chan RadioEvent, 10)
+	d := Device{
+		spi:            spi,
+		radioEventChan: c,
+	}
+	// On STM32WL SoC, SPI3 is an internal bus to sx1262 (subghz)
+	// radio. It requires specific bus initialization.
+	if d.spi == machine.SPI3 {
+		d.SubGhzInit()
+	}
+	return &d
+}
+
+//SpiTx Receive byte over SPI
+func (d *Device) SpiTx(cmd uint8) uint8 {
+	r, _ := d.spi.Transfer(cmd)
+	return r
+}
+
+//SpiRx Receive byte over SPI
+func (d *Device) SpiRx() uint8 {
+	r, _ := d.spi.Transfer(0xFF)
+	return r
+}
+
+//SpiSetNss Sets the NSS line
+func (d *Device) SpiSetNss(state bool) {
+	if d.spi != machine.SPI3 {
+		for {
+			// Only STM32WL's internal sx126x radio is yet implemented
+		}
+	}
+	if state {
+		stm32.PWR.SUBGHZSPICR.SetBits(stm32.PWR_SUBGHZSPICR_NSS)
+	} else {
+		stm32.PWR.SUBGHZSPICR.ClearBits(stm32.PWR_SUBGHZSPICR_NSS)
+
+	}
+}
+
+// WaitBusy sleep until all busy flags clears
+func (d *Device) WaitBusy() error {
+	count := 100
+	var rfbusyms, rfbusys bool
+	if d.spi != machine.SPI3 {
+		for {
+			// Only STM32WL's internal sx126x radio is yet implemented
```

Comment:
Same comment about infinite loops as above.

Reply:
I added a panic() in the constructor if SPI is not SPI3.